efsity/src/main/java/org/smartregister/command/PublishFhirResourcesCommand.java

```diff
@@ -171,32 +177,98 @@ void setProperties(Properties properties) {
     }
   }
 
-  void publishResources() throws IOException, ValidationException, GenerationException {
-    ArrayList<String> resourceFiles = getResourceFiles(projectFolder);
+  String getFileName(String name) {
+    if ((name.endsWith("Register")) || (name.endsWith("Profile"))) {
+      String regex = "([a-z])([A-Z]+)";
+      String replacer = "$1_$2";
+      name = name.replaceAll(regex, replacer).toLowerCase();
+    }
+    if (name.startsWith("strings")) {
+      name = name + "_config.properties";
+    } else {
+      name = name + "_config.json";
+    }
+    return name;
+  }
+
+  HashMap<String, String> getDetails(JSONObject jsonObject) {
+    JSONObject focus = jsonObject.getJSONObject("focus");
+    String reference = focus.getString("reference");
+    JSONObject identifier = focus.getJSONObject("identifier");
+    String name = identifier.getString("value");
+
+    HashMap<String, String> map = new HashMap<>();
+    map.put("reference", reference);
+    map.put("name", getFileName(name));
+    return map;
+  }
+
+  String getBinaryContent(String fileName, String projectFolder) throws IOException {
+    String pathToFile;
+    if (fileName.contains("register")) {
+      pathToFile = projectFolder + "/registers/" + fileName;
```

Comment:
assuming we're always using this agreed on file/folder structure

Reply:
yea seems reasonable, on that note, should we to the validation checker (and later the app content from template creator) something to verify the repo follows file structure?

Reply:
sure, created an issue here https://github.com/onaio/fhir-tooling/issues/195